tests/conftest.py

```diff
@@ -2988,8 +2988,10 @@ def get_ready_noobaa_endpoint_count(namespace):
     pods_info = get_pods_having_label(label=constants.NOOBAA_ENDPOINT_POD_LABEL, namespace=namespace)
     ready_count = 0
     for ep_info in pods_info:
-        if ep_info['status']['containerStatuses'][0]['ready']:
-            ready_count += 1
+        container_statuses = ep_info.get('status', {}).get('containerStatuses')
+        if container_statuses and container_statuses is not []:
+            if container_statuses[0].get('ready'):
```

Comment:
the above 'if' statement is if the container_statusses is not a list and here you are accessing it as a list.

Reply:
the condition is if it's not an empty list

Reply:
If `container_statuses` is `[]`, then `bool(container_statuses)` is false, so that the 2nd check isn't necessary. I would rather used something like `len(container_statuses) > 0` if the intention is to process nonempty lists.

Reply:
You can change:
if len(container_statuses)>0:
    if container_statuses[0].get('ready'):

Reply:
@mbukatov @OdedViner 
the second part of the condition (`if container_statuses`) is to for cases in which `container_statuses` is actually `None`.

but what if `container_statuses` is None?

```
>>> container_statuses = None
>>> len(container_statuses)>0
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    len(container_statuses)>0
TypeError: object of type 'NoneType' has no len()
```

Reply:
@OdedViner 

`container_statuses` might be `None`, empty lists, or a non-empty list. 
The condition should filter only non-empty lists

`len(arr) > 0` will not fit for the condition because `len(None)` will throw.

Which mean we need: `if container_statuses and len(container_statuses) > 0` which is eqivulent to the current condition:
`if container_statuses and container_statuses is not []` in the current context.

I think The condition is OK the way it is.

Reply:
Ah, right. In such case, it would be more readable if written as `if container_statuses is not None and len(container_statuses) > 0:` ...

Reply:
sure, will update

Reply:
done